platform/reworkd_platform/web/api/memory/memory_with_fallback.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+from typing import List
+
+from loguru import logger
+
+from reworkd_platform.web.api.memory.memory import AgentMemory
+
+
+class MemoryWithFallback(AgentMemory):
+    """
+    Wrap a primary AgentMemory provider and use a fallback in the case that it fails
+    We do this because we've had issues with Weaviate crashing and causing memory to randomly fail
+    """
+
+    def __init__(self, primary: AgentMemory, secondary: AgentMemory):
+        self.primary = primary
+        self.secondary = secondary
+
+    def __enter__(self) -> AgentMemory:
+        try:
+            return self.primary.__enter__()
+        except Exception as e:
+            logger.error(e)
```

Comment:
```suggestion
          logger.exception(e)
```

We will want to setup sentry to catch this eventually